src/llama_stack/providers/inline/agents/meta_reference/responses/streaming.py

```diff
@@ -272,7 +272,7 @@ async def create_response(self) -> AsyncIterator[OpenAIResponseObjectStream]:
 
                 # Handle choices with no tool calls
                 for choice in current_response.choices:
-                    if not (choice.message.tool_calls and self.ctx.response_tools):
+                    if not (isinstance(choice.message, OpenAIAssistantMessageParam) and choice.message.tool_calls and self.ctx.response_tools):
```

Comment:
i think this could be better readable perhaps with a `has_tool_call` boolean assignment and then test against that